packages/web/src/components/pages/TestSuites/TestSuiteDetails/TestSuiteSettings/SettingsTests/SettingsTests.tsx

```diff
@@ -92,7 +92,7 @@ const SettingsTests = () => {
 
   const [steps, setSteps] = useState(initialSteps);
 
-  const wasTouched = steps !== initialSteps;
+  const wasTouched = useMemo(() => steps !== initialSteps, [steps, initialSteps]);
```

Comment:
`useMemo` is unnecessary here - deep comparison is cheap, and will anyway be done here, multiple times.
```suggestion
  const wasTouched = steps !== initialSteps;
```